libtransact/src/protocol/receipt.rs

```diff
@@ -420,37 +461,37 @@ impl std::fmt::Display for TransactionReceiptBuilderError {
 }
 
 #[derive(Default, Clone)]
-pub struct TransactionReceiptBuilder {
+pub struct ValidTransactionReceiptBuilder {
```

Comment:
Shouldn't there be an analogous `InvalidTransactionReceiptBuilder`?  Or a single builder that would do something like:

```
TransactionReceiptBuilder::new()
    .valid()
    .with_state_changes(...)
   ...
```
and 
```
TransactionReceiptBuilder::new()
    .invalid()
    .with_message("Some message".to_string())
    ...
```

Reply:
Updated